lib/foreman_maintain/utils/backup.rb

```diff
@@ -213,16 +213,7 @@ def incremental?
       private
 
       def tarball_file_list(tarball)
-        # accepts tar.gz files only
-        file_list = []
-        File.open(tarball, 'rb') do |file|
-          ::Zlib::GzipReader.wrap(file) do |gz|
-            ::Gem::Package::TarReader.new(gz) do |tar|
-              tar.each { |entry| file_list << entry.full_name }
-            end
-          end
-        end
-        file_list
+        execute("tar -tf #{tarball}").split("\n")
```

Comment:
I think this method can be used when there are multiple config files to verify.
But this method is used only by `validate_hostname?` method to verify config file `etc/httpd/conf/httpd.conf` file exists or not. 

So why not to use `tar` command directly?
For example -  `execute?("tar -tf #{tarball}  #{file_name}")`

What do you think?

Reply:
I like the idea